analysis/queries.py

```diff
@@ -14,20 +14,22 @@
 """Database queries for acquiring experiment data."""
 
 import pandas as pd
-import sqlalchemy
 
-from database import models
+from database.models import Experiment, Trial, Snapshot
 from database import utils as db_utils
 
 
 def get_experiment_data(experiment_names):
     """Get measurements (such as coverage) on experiments from the database."""
-    snapshots_query = db_utils.query(models.Snapshot).options(
-        sqlalchemy.orm.joinedload('trial')).filter(
-            models.Snapshot.trial.has(
-                models.Trial.experiment.in_(experiment_names)))
 
-    # id must be loaded to do the join but get rid of it now since
-    # trial_id provides the same info.
-    data = pd.read_sql_query(snapshots_query.statement, db_utils.engine)
-    return data.drop(columns=['id'])
+    snapshots_query = db_utils.query(
+        Experiment.git_hash,\
+        Trial.experiment, Trial.fuzzer, Trial.benchmark,\
+        Trial.time_started, Trial.time_ended,\
+        Snapshot.trial_id, Snapshot.time, Snapshot.edges_covered)\
+        .select_from(Experiment)\
+        .join(Trial)\
+        .join(Snapshot)\
+        .filter(Experiment.name.in_(experiment_names))
```

Comment:
Are you sure you don't need to join on trial_id = id?
For example, this is how the query is done in SQL: `SELECT * FROM snapshot JOIN trial ON experiment = '2020-03-23' and id = trial_id ORDER BY time_ended;`
Maybe this isn't needed (as it wasn't in the last query) because SQLAlchemy knows the two fields are connected?

Also, we generally don't use backslashes to break up lines of code.

Reply:
> Maybe this isn't needed (as it wasn't in the last query) because SQLAlchemy knows the two fields are connected?

Yes, SQLAlchemy automatically constructs the `JOIN` conditions from the relationships encoded in the models. This is the generated SQL query:
```sql
SELECT experiment.git_hash AS experiment_git_hash, trial.experiment AS trial_experiment, trial.fuzzer AS trial_fuzzer, trial.benchmark AS trial_benchmark, trial.time_started AS trial_time_started, trial.time_ended AS trial_time_ended, snapshot.trial_id AS snapshot_trial_id, snapshot.time AS snapshot_time, snapshot.edges_covered AS snapshot_edges_covered 
FROM experiment JOIN trial ON experiment.name = trial.experiment JOIN snapshot ON trial.id = snapshot.trial_id 
WHERE experiment.name IN (%(name_1)s)
```

> Also, we generally don't use backslashes to break up lines of code.

I know, that's why I put it in a separate commit, see https://github.com/google/fuzzbench/pull/177/commits/c70211404c630c8e9b509f850ca91f73749d6d99. I think in this case it largely improves readability and yapf didn't complain either :) I'd keep it unless you feel strongly about it. Wdyt?

Reply:
OK fine to keep